obsidianirc.c

```diff
@@ -0,0 +1,640 @@
+    {
+        sendto_one(client, NULL, ":%s REGISTER SUCCESS %s :Account registered successfully.", me.name, name);
+        strlcpy(client->user->account, name, sizeof(client->user->account));
+        user_account_login(NULL, client);
+        RunHook(HOOKTYPE_ACCOUNT_REGISTER, acc, client);
+    }
+    else
+    {
+        sendto_one(client, NULL, ":%s FAIL REGISTER INTERNAL_ERROR :Failed to register account.", me.name);
+    }
+    free_account(acc);
+}
+
+/**
+ * list_accounts - Handles the LISTACC command from users.
+ */
+CMD_FUNC(list_accounts)
+{
+    Account **accounts = read_accounts_from_db(!BadPtr(parv[1]) ? parv[1] : NULL);
+    if (!accounts || !accounts[0])
+    {
+        sendto_one(client, NULL, ":%s LISTACC NO_ACCOUNTS :No accounts registered.", me.name);
+        if (accounts)
+        {
+            free(accounts);
+        }
+        return;
+    }
+    for (size_t i = 0; accounts[i]; i++)
+    {
+        // Count logged-in users (members)
+        int member_count = 0;
+        AccountMember *m = accounts[i]->members;
+        while (m)
+        {
+            member_count++;
+            m = m->next;
+        }
+        sendto_one(client, NULL, ":%s LISTACC ACCOUNT %ld %s %s %ld %d %d",
+            me.name,
+            accounts[i]->id,
+            accounts[i]->name,
+            accounts[i]->email,
+            (long)accounts[i]->time_registered,
+            accounts[i]->verified,
+            member_count // number of logged-in users
+        );
+        free_account(accounts[i]);
+    }
+    free(accounts);
+}
+
+/**
+ * my_find_tkl_nameban - Checks if a given name is banned via TKL nameban.
+ * Returns a pointer to the TKL if found, otherwise NULL.
+ */
+TKL *my_find_tkl_nameban(const char *name)
+{
+    TKL *tkl;
+
+    for (tkl = tklines[tkl_hash('Q')]; tkl; tkl = tkl->next)
+    {
+        if (!TKLIsNameBan(tkl))
+        {
+            continue;
+        }
+        if (!strcasecmp(name, tkl->ptr.nameban->name))
+        {
+            return tkl;
+        }
+    }
+    return NULL;
+}
+
+/**
+ * accreg_capability_parameter - Returns the parameter string for the registration capability.
+ */
+const char *accreg_capability_parameter(Client *client)
+{
+    return "before-connect,custom-account-name,email-required";
+}
+
+/**
+ * accreg_capability_visible - Determines if the registration capability is visible to the client.
+ */
+int accreg_capability_visible(Client *client)
+{
+    return 1;
+}
+
+/**
+ * open_database - Opens the SQLite database.
+ */
+int open_database(const char *filename)
+{
+    if (sqlite3_open(filename, &db) != SQLITE_OK)
+    {
+        return SQLITE_ERROR;
+    }
+    const char *sql =   "CREATE TABLE IF NOT EXISTS accounts ("
+                        "id INTEGER PRIMARY KEY AUTOINCREMENT, "
+                        "name TEXT, "
+                        "email TEXT, "
+                        "password TEXT, "
+                        "time_registered INTEGER, "
```

Comment:
You could have an automatic timestamp like 
```suggestion
                        "time_registered DATETIME DEFAULT CURRENT_TIMESTAMP,"
, "
```

There is also the proper datetime type

Not a huge deal